copy/copy.go

```diff
@@ -1060,6 +1060,14 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 			logrus.Debugf("Skipping blob %s (already present):", srcInfo.Digest)
 			bar := ic.c.createProgressBar(pool, srcInfo, "blob", "skipped: already exists")
 			bar.SetTotal(0, true)
+
+			// Throw an event that the layer has been skipped
+			if ic.c.progress != nil {
+				ic.c.progress <- types.ProgressProperties{
+					Event:    types.ProgressEventSkipped,
```

Comment:
(Should there be a `ProgressEventNewArtifact` event as well?

I’m don’t see a strong reason to prefer either way, maybe not doing it is a tiny bit simpler. I just want to make sure this has been considered.)

Reply:
We already have it here:
https://github.com/containers/image/blob/8d71f3f4c9aafaca1952fabfa51e329c3b7d46ed/copy/progress_reader.go#L32-L37

If you want then I could follow-up by adding a new event and changing the value there. The thing is that we will break the API with something like that.

Reply:
I meant: should the sequence of events for a skipped blob be `NewArtifact`+`Skipped`, or `Skipped` only?

I can imagine a caller that would “register” a blob on `NewArtifact` and “mark as done” on either `Skipped` or `Done`, where it would be convenient to emit both. (Hum, in that case we probably should call `NewArtifact` before `TryReusingBlob` spends time waiting for a remote registry.)

OTOH I can well imagine a different caller that wants to differentiate between skipped blobs (e.g. no progress bar at all, the roundtrips for reuse detection are considered negligible) and copied blobs, and in that case it’s useful that skipped blobs don’t start with `NewArtifact`; we would need a `NewArtifact`+`ReallyStartingToCopy`+`Read`+`Done`.

Looking at that `ReallyStartingToCopy`… let’s not worry about that now. We can, later, add a `CheckingForReuse` phase before the `NewArtifact`/`Skipped` report, if it turned out useful to watch for the time spent in `TryReuseBlob` .